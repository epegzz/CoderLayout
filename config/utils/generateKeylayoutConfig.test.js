const generateKeylayoutConfig = require('./generateKeylayoutConfig')

describe('generateKeylayoutConfig', () => {
  test('works', () => {
    const keylayoutKeyCodes = {
      'x': 100,
      'y': 101,
      'return_or_enter': 103
    }
    const keylayoutOutputMapping = {
      x: ['a', 'A'],
      y: ['b', 'B'],
      return_or_enter: 'return'
    }
    const keylayoutOutputAliases = {
      return: '&#x000D;'
    }

    expect(generateKeylayoutConfig({
      keylayoutKeyCodes,
      keylayoutOutputMapping,
      keylayoutOutputAliases
    })).toMatchSnapshot()
  })
})
