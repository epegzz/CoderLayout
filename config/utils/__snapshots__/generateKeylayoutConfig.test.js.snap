// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateKeylayoutConfig works 1`] = `
Array [
  Object {
    "encodedOutput": "&#x0061;",
    "output": "a",
    "virtualKeyCode": 100,
    "virtualKeyName": "x",
  },
  Object {
    "encodedOutput": "&#x0041;",
    "output": "A",
    "shift": true,
    "virtualKeyCode": 100,
    "virtualKeyName": "x",
  },
  Object {
    "encodedOutput": "&#x0062;",
    "output": "b",
    "virtualKeyCode": 101,
    "virtualKeyName": "y",
  },
  Object {
    "encodedOutput": "&#x0042;",
    "output": "B",
    "shift": true,
    "virtualKeyCode": 101,
    "virtualKeyName": "y",
  },
  Object {
    "encodedOutput": "&#x000D;",
    "output": "return",
    "virtualKeyCode": 103,
    "virtualKeyName": "return_or_enter",
  },
  Object {
    "encodedOutput": "&#x000D;",
    "output": undefined,
    "shift": true,
    "virtualKeyCode": 103,
    "virtualKeyName": "return_or_enter",
  },
]
`;
